scripts/mastodon_setup/provision/script/script.rb

```diff
@@ -34,33 +36,53 @@
 skytap_username = lab_control.find_metadata_attr('skytap_username')
 skytap_token = lab_control.find_metadata_attr('skytap_token')
 skytap_pip_interface_hostname = lab_control.find_metadata_attr('skytap_pip_interface_hostname')
-
+skytap_mastodon_server_template_id = lab_control.find_metadata_attr('skytap_mastodon_server_template_id')
 lab_uuid = lab_control.find_metadata_attr('lab_uuid')
 
-
 configuration_url = skytap_metadata.metadata['configuration_url']
 
 unless lab_uuid
   lab_uuid = SecureRandom.uuid
   lab_control.update_control_data({ "metadata" => { "lab_uuid" => lab_uuid }})
 end
 
-puts "Configuring lab public IP and DNS..."
-
 skytap_client = SkytapClient.new(skytap_username, skytap_token)
+original_skytap_environment = skytap_client.get(configuration_url)
+originally_running = original_skytap_environment['vms'].any? {|vm| vm['runstate'] == 'running'}
+
+puts "Adding Mastodon server to environment..."
```

Comment:
If the script is run multiple times, this merges a new VM in every time it runs. Should we check if a VM called "Mastodon" is already in the environment before doing this?

Reply:
Yes, good idea, will do